src/olympia/git/utils.py

```diff
@@ -1,796 +0,0 @@
-
-        self.git_repository_path = os.path.join(
-            settings.GIT_FILE_STORAGE_PATH,
-            id_to_path(self.addon_id, breadth=2),
-            package_type,
-        )
-
-    @property
-    def is_extracted(self):
-        return os.path.exists(self.git_repository_path)
-
-    @property
-    def is_recent(self):
-        git_description_path = os.path.join(
-            self.git_repository_path, self.GIT_DESCRIPTION
-        )
-        if not self.is_extracted or not os.path.exists(git_description_path):
-            return False
-        # A git repository is recent when it was created less than 1 hour ago.
-        an_hour_ago = datetime.utcnow() - timedelta(hours=1)
-        mtime = datetime.utcfromtimestamp(
-            os.path.getmtime(
-                # There is no way to get the creation time of a file/folder on
-                # most UNIX systems, so we use a file that is created by
-                # default but never modified (GIT_DESCRIPTION) to determine
-                # whether a git repository is recent or not.
-                git_description_path
-            )
-        )
-        return mtime > an_hour_ago
-
-    @cached_property
-    def git_repository(self):
-        if not self.is_extracted:
-            os.makedirs(self.git_repository_path)
-            git_repository = pygit2.init_repository(
-                path=self.git_repository_path, bare=False
-            )
-            # Write first commit to 'master' to act as HEAD
-            tree = git_repository.TreeBuilder().write()
-            git_repository.create_commit(
-                'HEAD',  # ref
-                self.get_author(),  # author, using addons-robot
-                self.get_author(),  # commiter, using addons-robot
-                'Initializing repository',  # message
-                tree,  # tree
-                [],
-            )  # parents
-
-            log.info('Initialized git repository "%s"', self.git_repository_path)
-        else:
-            git_repository = pygit2.Repository(self.git_repository_path)
-
-            # We have to verify that the 'master' branch exists because we
-            # might have mis-initialized repositories in the past.
-            # See: https://github.com/mozilla/addons-server/issues/14127
-            try:
-                master_ref = 'refs/heads/master'
-                git_repository.lookup_reference(master_ref)
-            except KeyError as exc:
-                message = f'Reference "{master_ref}" not found'
-                log.exception(message)
-                raise MissingMasterBranchError(message) from exc
-
-        return git_repository
-
-    def delete(self):
-        if not self.is_extracted:
-            log.error('called delete() on a non-extracted git repository')
-            return
-        # Reset the git hash of each version of the add-on related to this git
-        # repository.
-        Version.unfiltered.filter(addon_id=self.addon_id).update(git_hash='')
```

Comment:
We probably want to get rid of the `git_hash` field on the `Version` model too, eventually. Let's file a follow-up if we cannot do this here.

Reply:
We'll have to do it in 2 steps (first make it nullable, then eventually remove it) but I'd like to do it separately from this PR.